mpris-proxy: add module

ThibautMarty opened the pull request:
### Description

This adds a new module service running mpris-proxy from Bluez. It can be used to control media players with bluethooth headsets' buttons. I use it for a few months and it works well.
I welcome feedback for the systemd service (should we use `default.target`? Restart policy? etc).
I did not add tests as the only thing the module does is generating a systemd service (which I assume is well tested). There is no option, no configuration file, etc.

It is also my first new module to home-manager so please tell me if I did something wrong. I have a few more modules that I would like to upstream.
Please note that I'm listed as a maintainer in NixOS so I didn't add myself to home-manager's maintainers.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

berbiche commented on modules/services/mpris-proxy.nix:
```diff
@@ -0,0 +1,31 @@
+        PartOf = [ "default.target" ];
+      };
+
+      Install = { WantedBy = [ "default.target" ]; };
```
```suggestion
      Install.WantedBy = [ "bluetooth.target" ];
```

berbiche commented on modules/services/mpris-proxy.nix:
```diff
@@ -0,0 +1,31 @@
+
+  config = mkIf cfg.enable {
+    systemd.user.services.mpris-proxy = {
+      Unit = {
+        Description =
+          "Proxy forwarding bluetooth midi controls via mpris2 to control media players";
+        PartOf = [ "default.target" ];
+      };
```
```suggestion
      Unit = {
        Description =
          "Proxy forwarding bluetooth midi controls via mpris2 to control media players";
        BindsTo = [ "bluetooth.target" ];
        After = [ "bluetooth.target" ];
      };
```

berbiche reviewed (changes requested):
Thanks for the contribution.

I didn't know this existed until now :smile: 

In regards to `bluetooth.target`:
https://www.freedesktop.org/software/systemd/man/systemd.special.html#bluetooth.target

ThibautMarty commented:
Thanks, fixed and rebased on master.
The unit starts when bluetooth is turned on :+1: .

ThibautMarty commented:
Just changed a few things in the news file (and rebased on master):
```patch
diff --git c/modules/misc/news.nix i/modules/misc/news.nix
index b492daf..0ea0233 100644
--- c/modules/misc/news.nix
+++ i/modules/misc/news.nix
@@ -1881,9 +1881,10 @@ in
         '';
       }
       {
-        time = "2021-03-29T07:00:00+00:00";
+        time = "2021-04-23T10:00:00+00:00";
+        condition = hostPlatform.isLinux;
         message = ''
-          A new module is available: 'services.mpris-proxy'.
+          A new service is available: 'services.mpris-proxy'.
         '';
       }
     ];
```

sumnerevans commented on modules/services/mpris-proxy.nix:
```diff
@@ -0,0 +1,32 @@
+  meta.maintainers = [ maintainers.thibautmarty ];
+
+  options.services.mpris-proxy.enable = mkEnableOption
+    "a proxy forwarding bluetooth midi controls via mpris2 to control media players";
```
Technically, MIDI and MPRIS2 are abbreviations.

```suggestion
    "a proxy forwarding Bluetooth MIDI controls via MPRIS2 to control media players";
```

sumnerevans commented on modules/services/mpris-proxy.nix:
```diff
@@ -0,0 +1,32 @@
+    systemd.user.services.mpris-proxy = {
+      Unit = {
+        Description =
+          "Proxy forwarding bluetooth midi controls via mpris2 to control media players";
```
```suggestion
          "Proxy forwarding Bluetooth MIDI controls via MPRIS2 to control media players";
```

sumnerevans reviewed (changes requested):
Feel free to ignore these suggestions. They are very nit-picky.

berbiche reviewed (approved):
Thanks for the contribution!

The `news.nix` will have to be updated to fix the merge conflict.

ThibautMarty commented:
Done
